src/microbit/model/image.py

```diff
@@ -0,0 +1,268 @@
+import constants as CONSTANTS
+
+
+class Image:
+    def __init__(self, *args, **kwargs):
+
+        # Depending on the number of arguments
+        # in constructor, it treat args differently.
+
+        if len(args) == 0:
+            # default constructor
+            self.__LED = self.__string_to_array(CONSTANTS.BLANK)
+        elif len(args) == 1:
+            pattern = args[0]
+            if type(pattern) is str:
+                self.__LED = self.__string_to_array(pattern)
+            else:
+                raise TypeError("Image(s) takes a string")
+        else:
+
+            width = args[0]
+            height = args[1]
+
+            if width < 0 or height < 0:
+                # This is not in original, but ideally,
+                # image should fail non-silently
+                raise ValueError(CONSTANTS.INDEX_ERR)
+
+            if len(args) == 3:
+                # This option is for potential third bytearray arguments
+                byte_arr = args[2]
+                self.__LED = self.__bytes_to_array(width, height, byte_arr)
+            else:
+                self.__LED = self.__create_leds(width, height)
+
+    def width(self):
+        if len(self.__LED):

```

Comment:
since you check if self.__LED exists, shouldn't it be `if self.__LED`  ?

Reply:
this check sees if self.__LED is empty. This must be checked before we try to access its first element.

Reply:
Just FYI:
```
test_list = []
if test_list:
    print("This list has elements")
else:
    print("This list is empty")
```
The above code will print `This list is empty`.
I prefer the current check `len(self.__LED)`, but it's up to you.

Reply:
I see! Didn't notice you were accessing the first element :)